src/components/transaction-confirmation/transaction-confirmation.tsx

```diff
@@ -0,0 +1,218 @@
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+
+import {ethers} from 'ethers';
+import {View} from 'react-native';
+
+import {Color, getColor} from '@app/colors';
+import {
+  Button,
+  ButtonVariant,
+  DataView,
+  ISLMIcon,
+  PopupContainer,
+  Spacer,
+  Text,
+} from '@app/components/ui';
+import {createTheme} from '@app/helpers';
+import {useContacts, useTypedRoute, useUser, useWallet} from '@app/hooks';
+import {I18N} from '@app/i18n';
+import {Transaction} from '@app/models/transaction';
+import {EthNetwork} from '@app/services';
+import {WalletType} from '@app/types';
+
+interface TransactionConfirmationProps {
+  onDoneLedgerBt: (estimateFee: number) => void;
+  onDoneTransaction: (
+    transaction: ethers.providers.TransactionResponse,
+  ) => void;
+}
+
+export const TransactionConfirmation = ({
+  onDoneLedgerBt,
+  onDoneTransaction,
+}: TransactionConfirmationProps) => {
+  const providerId = useUser().providerId;
+  const {from, to, amount, fee, splittedTo} =
+    useTypedRoute<'transactionConfirmation'>().params;
+  const contacts = useContacts();
+  const wallet = useWallet(from);
+
+  const [estimateFee, setEstimateFee] = useState(fee ?? 0);
+  const [error, setError] = useState('');
+  const [disabled, setDisabled] = useState(false);
+
+  const contact = useMemo(() => contacts.getContact(to), [contacts, to]);
+
+  const onDone = useCallback(async () => {
+    if (wallet) {
+      if (wallet.type === WalletType.ledgerBt) {
+        onDoneLedgerBt(estimateFee);
+        return;
+      }
+
+      try {
+        setDisabled(true);
+
+        const ethNetworkProvider = new EthNetwork(wallet);
+
+        const transaction = await ethNetworkProvider.sendTransaction(
+          to,
+          amount,
+        );
+
+        if (transaction) {
+          Transaction.createTransaction(transaction, providerId, estimateFee);
+          onDoneTransaction(transaction);
+        }
+      } catch (e) {
+        console.log('onDone', e);
+        if (e instanceof Error) {
+          setError(e.message);
+        }
+      } finally {
+        setDisabled(false);
+      }
+    }
+  }, [
+    wallet,
+    to,
+    amount,
+    estimateFee,
+    providerId,
+    onDoneLedgerBt,
+    onDoneTransaction,
+  ]);
```

Comment:
In this case, there is no need to use `useCallback` at all, because we throw almost all states into the list of dependencies and use the function in 1 component

Reply:
fix all